process/preprocess/eventsPreProcessorV0.go

```diff
@@ -0,0 +1,131 @@
+package preprocess
+
+import (
+	"encoding/json"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	coreData "github.com/multiversx/mx-chain-core-go/data"
+	nodeData "github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/alteredAccount"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-notifier-go/data"
+)
+
+// TODO: dismiss this implementation after http integration is fully deprecated
+type eventsPreProcessorV0 struct {
+	*baseEventsPreProcessor
+}
+
+// NewEventsPreProcessorV0 will create a new events data preprocessor instance
+func NewEventsPreProcessorV0(args ArgsEventsPreProcessor) (*eventsPreProcessorV0, error) {
+	baseEventsPreProcessor, err := newBaseEventsPreProcessor(args)
+	if err != nil {
+		return nil, err
+	}
+
+	return &eventsPreProcessorV0{
+		baseEventsPreProcessor: baseEventsPreProcessor,
+	}, nil
+}
+
+// SaveBlock will handle the block info data
+func (d *eventsPreProcessorV0) SaveBlock(marshalledData []byte) error {
+	argsBlockS := struct {
```

Comment:
could have defined a structure that can also be used in testing

Reply:
i already have a struct like this but with interface field, here I need it to be with pointers; i would keep it as it is since this is the restored old implementation, and it will be removed later on anyway